exercises/tutorial_halfday/ex8_matrix-transpose-local-array.cpp

```diff
@@ -0,0 +1,348 @@
+  // 2) Index permutation
+  // 3) Dimensions of the array
+  //
+
+  using TILE_MEM =
+    RAJA::LocalArray<int, RAJA::Perm<0, 1>, RAJA::SizeList<TILE_DIM, TILE_DIM>>;
+
+  // **NOTE** Although the LocalArray is constructed
+  // the array memory has not been allocated.
+
+  TILE_MEM RAJA_Tile;
+
+  //--------------------------------------------------------------------------//
+  std::cout << "\n Running RAJA - sequential matrix transpose example ...\n";
+
+  std::memset(At, 0, N_r * N_c * sizeof(int));
+
+  ///
+  /// Exercise:
+  ///
+  ///   Implement the RAJA policy for a sequential matrix transpose with a local array.
+  ///   Use the statement InitLocalMem<RAJA::cpu_tile_mem, RAJA::ParamList<#>, ....>
+  ///   to allocate local array memory inside a kernel. The cpu_tile_mem policy
+  ///   specifies that memory should be allocated on the stack. The entries in the
+  ///   RAJA::ParamList identify RAJA local arrays in the parameter tuple to intialize.
+  ///
+  ///   Use the statement ForICount<N, RAJA::statement::Param<T>, RAJA::loop_exec,
+  ///   to extract the tile local index of the Nth segment. The resulting index is
+  ///   stored in location T of the parameter tuple.
+  ///
+
+  /*
+  RAJA::kernel_param<SEQ_EXEC_POL>( RAJA::make_tuple(RAJA::RangeSegment(0, N_c),
+                                                     RAJA::RangeSegment(0, N_r)),
+
+    RAJA::make_tuple((int)0, (int)0, RAJA_Tile),
+
+    [=](int col, int row, int tx, int ty, TILE_MEM &RAJA_Tile) {
+      RAJA_Tile(ty, tx) = Aview(row, col);
+    },
+
+    [=](int col, int row, int tx, int ty, TILE_MEM &RAJA_Tile) {
+      Atview(col, row) = RAJA_Tile(ty, tx);
+
+  });
+  */
+
+  checkResult<int>(Atview, N_c, N_r);
+  // printResult<int>(Atview, N_c, N_r);
+
+#if defined(RAJA_ENABLE_OPENMP)
+  //--------------------------------------------------------------------------//
+  std::cout << "\n Running RAJA - OpenMP (parallel outer loop) matrix "
+               "transpose example ...\n";
+
+  std::memset(At, 0, N_r * N_c * sizeof(int));
+
+  ///
+  /// Exercise:
+  ///
+  ///   Implement the RAJA policy for an OpenMP matrix transpose with a local array.
+  ///   Use the statement InitLocalMem<RAJA::cpu_tile_mem, RAJA::ParamList<#>, ....>
+  ///   to allocate local array memory inside a kernel. The cpu_tile_mem policy
+  ///   specifies that memory should be allocated on the stack. The entries in the
+  ///   RAJA::ParamList identify RAJA local arrays in the parameter tuple to intialize.
+  ///
+  ///   Use the statement ForICount<N, RAJA::statement::Param<T>, RAJA::loop_exec,
+  ///   to extract the tile local index of the Nth segment. The resulting index is
+  ///   stored in location T of the parameter tuple.
+  ///
+
+  /*
+  RAJA::kernel_param<OPENMP_EXEC_POL>(
+      RAJA::make_tuple(RAJA::RangeSegment(0, N_c), RAJA::RangeSegment(0, N_r)),
+      RAJA::make_tuple((int)0, (int)0, RAJA_Tile),
+
+      [=](int col, int row, int tx, int ty, TILE_MEM &RAJA_Tile) {
```

Comment:
Please change variables tx, ty to tcol, trow (here and elsewhere) so it's clearer.